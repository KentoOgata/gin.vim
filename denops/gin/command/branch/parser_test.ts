import { assertSnapshot } from "https://deno.land/std@0.200.0/testing/snapshot.ts";
import { parse } from "./parser.ts";

Deno.test("parse", async function (t): Promise<void> {
  await t.step("returns proper branches", async (t) => {
    const content = await Deno.readTextFile(
      new URL("./testdata/branch.txt", import.meta.url),
    );
    const result = parse(content.split("\n"));
    await assertSnapshot(t, result);
  });
});
