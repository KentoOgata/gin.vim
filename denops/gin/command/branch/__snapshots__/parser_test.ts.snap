export const snapshot = {};

snapshot[`parse > returns proper branches 1`] = `
{
  branches: [
    {
      active: false,
      branch: "gin-blame",
      commit: "b2a1c55",
      kind: "local",
      message: "WIP",
      record: "  gin-blame          b2a1c55 [main: ahead 1, behind 5] WIP",
      target: "gin-blame",
      upstream: "main",
    },
    {
      active: true,
      branch: "main",
      commit: "227f6c0",
      kind: "local",
      message: "Merge pull request #102 from lambdalisue/upgrade-git-browse",
      record: "* main               227f6c0 [origin/main] Merge pull request #102 from lambdalisue/upgrade-git-browse",
      target: "main",
      upstream: "origin/main",
    },
    {
      active: false,
      branch: "support-bang-bk",
      commit: "de1f346",
      kind: "local",
      message: ":+1: A bang (!) to forcibly open a buffer",
      record: "  support-bang-bk    de1f346 [support-bang: gone] :+1: A bang (!) to forcibly open a buffer",
      target: "support-bang-bk",
      upstream: "support-bang",
    },
    {
      active: false,
      branch: "support-read",
      commit: "ecf6a74",
      kind: "local",
      message: ":+1: Add \`postProcessor\` options",
      record: "  support-read       ecf6a74 [origin/support-read: gone] :+1: Add \`postProcessor\` options",
      target: "support-read",
      upstream: "origin/support-read",
    },
    {
      active: false,
      branch: "support-references",
      commit: "f1d2830",
      kind: "local",
      message: ":shower: Remove unused variables",
      record: "  support-references f1d2830 [origin/support-references: ahead 12, behind 8] :shower: Remove unused variables",
      target: "support-references",
      upstream: "origin/support-references",
    },
    {
      active: false,
      branch: "upgrade-deps",
      commit: "3ecec96",
      kind: "local",
      message: ":herb: Force a custom serializer of \`assertSnapshot\`",
      record: "  upgrade-deps       3ecec96 [origin/upgrade-deps: gone] :herb: Force a custom serializer of \`assertSnapshot\`",
      target: "upgrade-deps",
      upstream: "origin/upgrade-deps",
    },
    {
      active: false,
      branch: "upgrade-git-browse",
      commit: "3a534bc",
      kind: "local",
      message: ":package: Upgrade git-browse",
      record: "  upgrade-git-browse 3a534bc [origin/upgrade-git-browse: gone] :package: Upgrade git-browse",
      target: "upgrade-git-browse",
      upstream: "origin/upgrade-git-browse",
    },
  ],
}
`;
